src/librustdoc/html/sources.rs

```diff
@@ -37,8 +38,8 @@ impl<'a> DocFolder for SourceCollector<'a> {
         if self.scx.include_sources
             // skip all synthetic "files"
             && item.source.filename.is_real()
-            // skip non-local items
-            && item.def_id.is_local()
+            // skip non-local files
+            && item.source.cnum == LOCAL_CRATE
         {
             // If it turns out that we couldn't read this file, then we probably
             // can't read any of the files (generating html output from json or
```

Comment:
Do you want to close #70757 without fixing the thing where `include_sources` would be set to `false`? I guess it might be hard to actually support only being able to emit *some* of the files, and is much less relevant now.

Feel free to leave a FIXME comment and open an issue about it.

Reply:
Yeah, I guess we can leave #70757 open to track removing this error case.